pkg/integrations/github/ghrunner.go

```diff
@@ -1,104 +1,118 @@
-./config.cmd --token $ghToken --url %s --name %s --unattended --runasservice --replace %s`
-
-	// whitespace at the start is required since this is expanded in a cloud-init yaml file
-	// to start as service need to relable the runsvc.sh file on rhel: https://github.com/actions/runner/issues/3222
-	installActionRunnerSnippetLinux string = `  mkdir ~/actions-runner && cd ~/actions-runner` + "\n" +
-		`      curl -o actions-runner-linux.tar.gz -L %s` + "\n" +
-		`      tar xzf ./actions-runner-linux.tar.gz` + "\n" +
-		`      sudo ./bin/installdependencies.sh` + "\n" +
-		`      ./config.sh --token %s --url %s --name %s --unattended --replace %s` + "\n" +
-		`      sudo ./svc.sh install` + "\n" +
-		`      chcon system_u:object_r:usr_t:s0 $(pwd)/runsvc.sh` + "\n" +
-		`      sudo ./svc.sh start`
-
-	installActionRunnerSnippetMacos string = `mkdir ~/actions-runner && cd ~/actions-runner
-curl -o actions-runner-osx.tar.gz -L %s
-tar xzf ./actions-runner-osx.tar.gz
-./config.sh --token %s --url %s --name %s --unattended --replace %s
-./svc.sh install
-plistName=$(basename $(./svc.sh status | grep "plist$"))
-mkdir -p ~/Library/LaunchDaemons
-mv ~/Library/LaunchAgents/"${plistName}" ~/Library/LaunchDaemons/"${plistName}"
-launchctl load ~/Library/LaunchDaemons/"${plistName}"`
-)
+type snippetDataValues struct {
+	Username  string
+	Token     string
+	RepoURL   string
+	Name      string
+	Labels    string
+	RunnerURL string
+}
 
-var args *GithubRunnerArgs
+var runnerArgs *GithubRunnerArgs
 
-func InitGHRunnerArgs(gra *GithubRunnerArgs) error {
-	if gra.Token == "" || gra.Name == "" || gra.RepoURL == "" {
-		return errors.New("All args are required and must have non-empty values")
-	}
-	args = gra
-	return nil
+func Init(args *GithubRunnerArgs) {
+	runnerArgs = args
 }
 
-func GetToken() string {
-	var token = func() string {
-		return args.Token
+func SelfHostedRunnerSnippet(username string) (*string, error) {
+	if runnerArgs == nil {
+		noSnippet := ""
+		return &noSnippet, nil
 	}
-	return util.IfNillable(args != nil, token, "")
+	templateConfig := string(snippets[*runnerArgs.Platform][:])
+	snippet, err := file.Template(
+		snippetDataValues{
+			Name:      runnerArgs.Name,
+			Token:     runnerArgs.Token,
+			Labels:    GetLabels(),
+			RepoURL:   runnerArgs.RepoURL,
+			RunnerURL: downloadURL(),
+			Username:  username,
+		},
+		templateConfig)
+	return &snippet, err
 }
 
-func GetLabels() string {
-	var labels = func() string {
-		if len(args.Labels) > 0 {
-			return fmt.Sprintf("--labels %s", strings.Join(args.Labels, ","))
-		}
-		return ""
+// If we add the snippet as part of a cloud init file the strategy
+// would be create the file with write_files:
+// i.e.
+// write_files:
+//
+//	# Cirrus service setup
+//	- content: |
+//	    {{ .CirrusSnippet }} <----- 6 spaces
+//
+// to do so we need to indent 6 spaces each line of the snippet
+func SelfHostedRunnerSnippetAsCloudInitWritableFile(username string) (*string, error) {
```

Comment:
This functions is the same as we got on Cirrus? can we move it to a util pkg or something?

Reply:
ok let me know once you change this so I will approve

Reply:
@adrianriobo updated the PR, i followed a slightly different approach at the end, PTALA